cve_bin_tool/checkers/firefox.py

```diff
@@ -0,0 +1,23 @@
+# Copyright (C) 2022 Orange
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+
+"""
+CVE checker for firefox
+
+https://www.cvedetails.com/product/3264/Mozilla-Firefox.html?vendor_id=452
+
+"""
+from __future__ import annotations
+
+from cve_bin_tool.checkers import Checker
+
+
+class FirefoxChecker(Checker):
+    CONTAINS_PATTERNS: list[str] = []
+    FILENAME_PATTERNS: list[str] = [r"firefox"]
+    VERSION_PATTERNS = [
+        r"ENCODER=Mozilla([0-9]+.[0-9]+(.[0-9]+)?).*Firefox",
+        r"{\"type\"\:\"rpm\",\"name\"\:\"firefox\"\,\"version\"\:\"10([0-9]+.[0-9]+.[0-9])-1.fc3([0-9])\"\,\"architecture\"\:\"x86_64\"\,\"osCpe\"\:\"cpe:/o:fedoraproject:fedora:38\"}",
```

Comment:
```suggestion
        r'"name":"firefox","version":"([0-9]+.[0-9]+(.[0-9]+)?)',
```

This would remove quite a lot of escaping backslash and will make it a lot readable. The numbers at the end like fedora:38 won't be consistent across packages. So we can remove that part of the string and still have a unique string in our hands.

I've changed the version pattern from `10([0-9]+.[0-9]+.[0-9])` to `([0-9]+.[0-9]+(.[0-9]+)?)` since

- the former only captures wrong version eg: 106.0.1 is captured as 6.0.1. The reason is the capturing group (the parenthesis) doesn't have the 10 captured since it's specified outside.
- The last part of the version string is optional as firefox also has versions like 106.0 as a valid version.

This will fix one of the test failure.